src/index.ts

```diff
@@ -46,51 +47,122 @@ declare global {
+    if (modeState === "") {
+        window.addEventListener("keypress", keyPressHandler);
+        canvas.addEventListener("mousedown", mouseDownHandler);
+        canvas.addEventListener("mousemove", mouseMoveHandler);
+        canvas.addEventListener("mouseup", mouseUpHandler);
+        canvas.addEventListener("mouseout", mouseOutHandler);
     }
-    canvas.addEventListener("mousedown", cutHandler);
-    canvas.addEventListener("mousemove", cutHandler);
-    canvas.addEventListener("mouseup", cutHandler);
-    canvas.addEventListener("mouseout", cutHandler);
+    modeState = "cutMode";
 }
 
 /**
- * A function to begin atom creation.
- * If ellipseMode was previously active, remove the listener.
+ * If there is no current mode creates the listeners.
+ * Sets the current mode to atom mode.
  */
 function atomMode() {
-    if (modeState !== "atomMode") {
-        removeListeners();
-        modeState = "atomMode";
+    if (modeState === "") {
+        window.addEventListener("keypress", keyPressHandler);
+        canvas.addEventListener("mousedown", mouseDownHandler);
+        canvas.addEventListener("mousemove", mouseMoveHandler);
+        canvas.addEventListener("mouseup", mouseUpHandler);
+        canvas.addEventListener("mouseout", mouseOutHandler);
     }
-    window.addEventListener("keypress", atomHandler);
-    canvas.addEventListener("mousedown", atomHandler);
-    canvas.addEventListener("mousemove", atomHandler);
-    canvas.addEventListener("mouseup", atomHandler);
-    canvas.addEventListener("mouseout", atomHandler);
+    modeState = "atomMode";
 }
 
 /**
- * Removes all listeners added in a certain mode.
+ * Calls the respective keypress function depending on current mode.
+ * @param event The event of a keypress
  */
-function removeListeners() {
-    if (modeState === "ellipseMode") {
-        canvas.removeEventListener("mousedown", cutHandler);
-        canvas.removeEventListener("mousemove", cutHandler);
-        canvas.removeEventListener("mouseup", cutHandler);
-        canvas.removeEventListener("mouseout", cutHandler);
-    } else if (modeState === "atomMode") {
-        window.removeEventListener("keypress", atomHandler);
-        canvas.removeEventListener("mousedown", atomHandler);
-        canvas.removeEventListener("mousemove", atomHandler);
-        canvas.removeEventListener("mouseup", atomHandler);
-        canvas.removeEventListener("mouseout", atomHandler);
+function keyPressHandler(event: KeyboardEvent) {
+    switch (modeState) {
+        case "atomMode":
+            atomKeyPress(event);
+            break;
+    }
+}
+
+/**
+ * Calls the respective mouse down function depending on current mode.
+ * @param event The mouse down event
+ */
+function mouseDownHandler(event: MouseEvent) {
+    switch (modeState) {
+        case "cutMode":
+            cutMouseDown(event);
+            hasMouseDown = true;
+            break;
+        case "atomMode":
+            hasMouseDown = atomMouseDown(event);
```

Comment:
The mouse is down now regardless of any logic going on in atomMode, why would we depend on this?

Reply:
I didn't have this initially and I found that since the logic for hasKeyPress is located in atomMode, it would assume mouseDown had fired successfully even if it lacked a proper letter to place creating blank atoms. I'll look for a better solution for this.

Reply:
Yea the hasMouseDown global should always be strongly linked to the real state of the mouse the mouse being down to avoid any confusion. If the mouse is down, it should be true and if the mouse isn't down (no button is being pressed) it should be true.

Reply:
Ok, that makes more sense than what I was doing thank you.